// dom-testing-library/dist/wait-for-dom-change.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.waitForDomChange = undefined;

var _helpers = require('./helpers');

function waitForDomChange() {
  var _ref = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {},
      _ref$container = _ref.container,
      container = _ref$container === undefined ? (0, _helpers.getDocument)() : _ref$container,
      _ref$timeout = _ref.timeout,
      timeout = _ref$timeout === undefined ? 4500 : _ref$timeout,
      _ref$mutationObserver = _ref.mutationObserverOptions,
      mutationObserverOptions = _ref$mutationObserver === undefined ? {
    subtree: true,
    childList: true,
    attributes: true,
    characterData: true
  } : _ref$mutationObserver;

  return new Promise(function (resolve, reject) {
    var timer = setTimeout(function () {
      onDone(new Error('Timed out in waitForDomChange.'), null);
    }, timeout);
    var observer = (0, _helpers.newMutationObserver)(function (mutationsList) {
      onDone(null, mutationsList);
    });
    observer.observe(container, mutationObserverOptions);

    function onDone(error, result) {
      clearTimeout(timer);
      setImmediate(function () {
        return observer.disconnect();
      });
      if (error) {
        reject(error);
      } else {
        resolve(result);
      }
    }
  });
}

exports.waitForDomChange = waitForDomChange;
